### DISCUSSION TITLE
wagmi core write contract

### DISCUSSION THREAD

**Discussion by Rickshanelorie** [wevm/wagmi#2940] [Q&A]
for wagmi core write contract how to get the confirmation on the transaction to show the snackbar after it fully sent?

**Comment by Ruithlzz09** [comment]
@Rickshanelorie 
If you can txnHash and want to wait for a certain duration before informing the user whether the current status or check later you can do something like this

**await waitForTransaction({chainId,timeout:30000,hash:payload.txHash})**


Ref: https://wagmi.sh/core/actions/waitForTransaction
